leader-election-impl/src/main/java/com/palantir/paxos/VerifyingPaxosStateLog.java

```diff
@@ -175,12 +201,15 @@ private long getExtremeLogEntry(Function<PaxosStateLog<V>, Long> extractor) {
         lock.readLock().lock();
         try {
             Long result = extractor.apply(currentLog);
-            Long experimentalResult = safeReadFromExperimental(extractor::apply);
-            if (!Objects.equals(result, experimentalResult) && result != PaxosAcceptor.NO_LOG_ENTRY) {
-                log.error("Mismatch in getting the extreme log entry between legacy and current implementations."
-                                + " Legacy result {}, current result {}.",
-                        SafeArg.of("legacy", result),
-                        SafeArg.of("current", experimentalResult));
+            if (firstEntry.isDone()) {
+                Long experimentalResult = safeReadFromExperimental(extractor::apply);
+                if (!Objects.equals(result, experimentalResult) && result != PaxosAcceptor.NO_LOG_ENTRY
+                        && result >= Futures.getUnchecked(firstEntry)) {
```

Comment:
```suggestion
                        && result > Futures.getUnchecked(firstEntry)) {
```

I think this is needed, otherwise we might end up claiming assertions on the minimum after writing just the very first entry